components/create_swagger_config.py

```diff
@@ -138,7 +138,15 @@ def find_best_type_for_prop(prop):
 
         elif one_type != 'null':
             prop['type'] = one_type
-            break
+            # prefer arrays first and strings second
+            if one_type == 'array' or (one_type == 'string' and
+                                       'items' not in prop):
+                break
```

Comment:
Does this actually make arrays and strings equally preferred?

Here's what I think I see: We're iterating over multiple_types, and we break if we find either array or string. So if we find a string first, we'll immediately break and potentially miss a later, preferred array.

Reply:
I can update the comment to make this a bit more clear, because even I wondered if it was correct after looking back at it. There are a few cases in the describe messages where a property can be either an array or a string, but in all of these cases, the property also has an `items` key in order to define the items in the array. Thus, I will give an example of where the array is preferred.

With the following property, the `prop['type']` will be assigned `array`, rather than `string`:
```
{
    'items': {'type': 'integer', 'description': 'The lnn of the node.'}, 
    'description': 'List of the nodes or "all" where the last upgrade action can be retried.', 
    'type': ['string', 'array']
}
```

Reply:
That makes sense. Thanks for the explanation.

Reply:
Thanks for the new comment, it makes it much clearer.